blockchain/stake/staketx_test.go

```diff
@@ -1326,6 +1353,144 @@ func TestIsNullDataScript(t *testing.T) {
+		PkScript: hexToBytes("ba76a914097e847d49c6806f6933e806a350f43b97ac70d088a" +
+			"c"),
+		Value:   4126629682,
+		Version: 0,
+	}
+	ticketOut2 := &MinimalOutput{
+		PkScript: hexToBytes("6a1e86c6da62556f5e21fbce3564b7374724d65f0cbb51c66d0" +
+			"0000000800058"),
+		Value:   0,
+		Version: 0,
+	}
+	ticketOut3 := &MinimalOutput{
+		PkScript: hexToBytes("bd76a914000000000000000000000000000000000000000088a" +
+			"c"),
+		Value:   0,
+		Version: 0,
+	}
+	ticketOut4 := &MinimalOutput{
+		PkScript: hexToBytes("6a1e7e8efe653b374ba7ad950c873e483fc53c5bafa9c1c439f" +
+			"8000000000058"),
+		Value:   0,
+		Version: 0,
+	}
+	ticketOut5 := &MinimalOutput{
+		PkScript: hexToBytes("bd76a914000000000000000000000000000000000000000088a" +
+			"c"),
+		Value:   0,
+		Version: 0,
+	}
+	ticketMinOuts := []*MinimalOutput{
+		ticketOut1,
+		ticketOut2,
+		ticketOut3,
+		ticketOut4,
+		ticketOut5,
+	}
+	revocationHash := mustParseHash("46ae5f78174c6c6e3675d0bbfec27e25c40f3a119e" +
+		"df9183b96261db5cda7a4f")
+	revocationTxFee := dcrutil.Amount(285000)
+	revocationTxVersion := uint16(1)
+
+	// Invalid ticket commitment amount.
+	ticketOutInvalidCommitmentAmt := &MinimalOutput{
+		PkScript: hexToBytes("6a1e86c6da62556f5e21fbce3564b7374724d65f0cbbfffffff" +
+			"fffffffff0058"),
+		Value:   0,
+		Version: 0,
+	}
+	ticketMinOutsInvalidCommitmentAmt := []*MinimalOutput{
+		ticketOut1,
+		ticketOutInvalidCommitmentAmt,
+		ticketOut3,
+		ticketOut4,
+		ticketOut5,
+	}
+
+	tests := []struct {
+		name                string
+		ticketHash          *chainhash.Hash
+		ticketMinOuts       []*MinimalOutput
+		revocationTxFee     dcrutil.Amount
+		revocationTxVersion uint16
+		wantTxHash          chainhash.Hash
+		wantErr             error
+	}{{
+		name:                "valid with P2SH and P2PKH outputs",
+		ticketHash:          ticketHash,
+		ticketMinOuts:       ticketMinOuts,
+		revocationTxFee:     revocationTxFee,
+		revocationTxVersion: revocationTxVersion,
+		wantTxHash:          *revocationHash,
+	}, {
+		name:                "invalid ticket commitment amount",
+		ticketHash:          ticketHash,
+		ticketMinOuts:       ticketMinOutsInvalidCommitmentAmt,
+		revocationTxFee:     revocationTxFee,
+		revocationTxVersion: revocationTxVersion,
+		wantErr:             ErrSStxBadCommitAmount,
+	}}
+
+	for _, test := range tests {
+		// Create a revocation transaction with the given test parameters.
+		revocationTx, err := CreateRevocationFromTicket(test.ticketHash,
+			test.ticketMinOuts, test.revocationTxFee, test.revocationTxVersion,
+			params)
+
+		// Validate that the expected error was returned for negative tests.
+		if test.wantErr != nil {
+			if !errors.Is(err, test.wantErr) {
+				t.Errorf("%q: mismatched error -- got %T, want %T", test.name, err,
+					test.wantErr)
+			}
+			continue
+		}
+
+		// Validate that an unexpected error was not returned.
+		if err != nil {
+			t.Fatalf("%q: unexpected error creating revocation: %v", test.name, err)
+		}
```

Comment:
This is mostly a stylistic choice (so feel free not to change it), but you can also do the inner test directly, since `errors.Is(nil, nil) == true`:

```go
if !errors.Is(err, test.wantErr) {
	t.Fatalf("%q: mismatched error -- got %T, want %T", test.name, err,
		test.wantErr)
}

if test.wantErr != nil {
	// Negative tests don't produce a revocation that can be further checked.
	continue
}
```

https://play.golang.org/p/XvIeJgNybIl

Reply:
I actually had something similar at first, but there is one difference.  If you fail a positive test, it will fall into that first block with your version, which produces an error message that is less descriptive than if it falls into the unexpected error case, which includes the full error string.